src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -1646,12 +1653,33 @@ private static bool LoadPSReadline()
         /// Opens and Initializes the Host's sole Runspace.  Processes the startup scripts and runs any command passed on the
         /// command line.
         /// </summary>
-        private void DoCreateRunspace(string initialCommand, bool skipProfiles, bool staMode, string configurationName, Collection<CommandParameter> initialCommandArgs)
+        /// <param name="args">Runspace creation event arguments.</param>
+        private void DoCreateRunspace(RunspaceCreationEventArgs args)
         {
-            Dbg.Assert(_runspaceRef == null, "runspace should be null");
+            Dbg.Assert(_runspaceRef == null, "_runspaceRef field should be null");
             Dbg.Assert(DefaultInitialSessionState != null, "DefaultInitialSessionState should not be null");
             s_runspaceInitTracer.WriteLine("Calling RunspaceFactory.CreateRunspace");
 
+            // Use session configuration file if provided.
+            bool customConfigurationProvided = false;
+            if (!string.IsNullOrEmpty(args.ConfigurationFilePath))
+            {
+                try
+                {
+                    // Replace DefaultInitialSessionState with the initial state configuration defined by the file.
+                    DefaultInitialSessionState = InitialSessionState.CreateFromSessionConfigurationFile(
```

Comment:
I think having that API to throw in case `-ConfigurationFile` is specified is a better behavior. Otherwise, the user would assume the passed-in `InitialSessionState` instance is used to create the Runspace.